library/core/src/ptr/const_ptr.rs

```diff
@@ -444,9 +444,11 @@ impl<T: ?Sized> *const T {
     ///
     /// `count` is in units of **bytes**.
     ///
-    /// This is purely a convenience for casting to a `u8` pointer and
-    /// using [offset][pointer::offset] on it. See that method for documentation
-    /// and safety requirements.
+    /// This is purely a convenience for casting to a `u8` pointer and using
+    /// [offset][pointer::offset] on it. See that method for documentation and
+    /// safety requirements. Note that the usual guidance on operations on
+    /// zero-sized referents does not apply here — the only valid `count` for
+    /// zero-sized referents is `0`.
```

Comment:
The docs for pointer::offset don't even have any guidance about zero-sized references? In fact, neither operation involves any reference at all! Only raw pointers are involved.

I don't understand which problem you see with the current docs, and I think the new sentence is quite confusing.

Reply:
So to be clear, please undo all these changes to the bytewise offset methods in your PR, and only keep the change in `library/core/src/ptr/mod.rs`.